src/rcheevos/rc_client.c

```diff
@@ -3345,6 +3423,62 @@ static void rc_client_submit_leaderboard_entry_retry(rc_client_scheduled_callbac
   rc_client_submit_leaderboard_entry_server_call(lboard_data);
 }
 
+static void rc_client_raise_scoreboard_event(rc_client_submit_leaderboard_entry_callback_data_t* lboard_data,
+    const rc_api_submit_lboard_entry_response_t* response)
+{
+  rc_client_leaderboard_scoreboard_t sboard;
+  rc_client_event_t client_event;
+  rc_client_subset_info_t* subset;
+  rc_client_t* client = lboard_data->client;
+  const rc_client_leaderboard_info_t* leaderboard = NULL;
+
+  if (!client || !client->game)
+    return;
+
+  for (subset = client->game->subsets; subset; subset = subset->next) {
+    leaderboard = rc_client_subset_get_leaderboard_info(subset, lboard_data->id);
+    if (leaderboard != NULL)
+      break;
+  }
+  if (leaderboard == NULL) {
+    RC_CLIENT_LOG_ERR_FORMATTED(client, "Trying to raise scoreboard for unknown leaderboard %u", lboard_data->id);
+    return;
+  }
+
+  sboard.leaderboard_id = lboard_data->id;
+  rc_format_value(sboard.submitted_score, sizeof(sboard.submitted_score), response->submitted_score, leaderboard->format);
+  rc_format_value(sboard.best_score, sizeof(sboard.best_score), response->best_score, leaderboard->format);
+  sboard.new_rank = response->new_rank;
+  sboard.num_entries = response->num_entries;
+  sboard.top_entries = NULL;
```

Comment:
I prefer `memset`ing the entire structure to zero in case more fields are added later.